src/vivarium_gates_mncnh/data/hemoglobin_effects/hgb_lbwsg_effect_generation.py

```diff
@@ -0,0 +1,218 @@
+        art_exposure['ga_start'][i] = art_cats[art_exposure.parameter[i]].split('- [')[1].split(', ')[0]
+        art_exposure['ga_end'][i] = art_cats[art_exposure.parameter[i]].split('- [')[1].split(') wks')[0].split(', ')[1]
+    art_exposure['ga_end'] = np.where(art_exposure.ga_end=='42+] wks', '42', art_exposure.ga_end)
+    art_exposure['bw_end'] = np.where(art_exposure.bw_end=='9999] g',1000,art_exposure.bw_end)
+    for col in ['ga_end','ga_start','bw_end','bw_start']:
+        art_exposure[f'{col}'] = art_exposure[f'{col}'].astype(int)
+    return art_exposure
+
+def get_lbwsg_birth_exposure(location):
+    # this will return birth exposure data data for both sexes and all draws for the specified location
+    # it will also include metadata for the exposure categories (GA/BW start/end values)
+
+    # TODO: update this to be relative to most recent artifact?
+    # Or pull directly using repo code?
+    artifact_dir = '/mnt/team/simulation_science/pub/models/vivarium_gates_mncnh/artifacts/model10.0/'
+    art = Artifact(artifact_dir + location + '.hdf')
+    exp = art.load('risk_factor.low_birth_weight_and_short_gestation.birth_exposure').reset_index()
+    art_exposure = get_lbwsg_metadata()
+    exp = exp.merge(art_exposure, on='parameter')
+    exp = (exp.set_index([x for x in exp.columns if 'draw' not in x])
+        .stack().reset_index()
+        .rename(columns={0:'exposure',
+                            'level_8':'draw',
+                            'sex_of_child':'sex'}))
+    exp['location'] = location
+    return exp
+
+def calculate_shift_from_rr(exp_data, rr_data, location, outcome, sex, draw, exposure_level):
+    """Calculates the shift in gestational age or birth weight exposure needed to achieve a target relative risk."""
+    if outcome == 'ptb':
+        exp_sub = exp_data.loc[(exp_data.sex==sex)&(exp_data.draw==draw)].set_index([x for x in exp_data.columns if 'ga' not in x])
+    elif outcome == 'lbw':
+        exp_sub = exp_data.loc[(exp_data.sex==sex)&(exp_data.draw==draw)].set_index([x for x in exp_data.columns if 'bw' not in x])
+    exp_tmrel = exp_sub.reset_index()
+    exp_tmrel['frac_preterm'] = np.where(exp_tmrel.ga_end <=37, 1,
+                                        np.where(exp_tmrel.ga_start >=37, 0,
+                                                (37 - exp_tmrel.ga_start)/(exp_tmrel.ga_end - exp_tmrel.ga_start)))
+    exp_tmrel['frac_lbw'] = np.where(exp_tmrel.bw_end <= 2500, 1,
+                                       np.where(exp_tmrel.bw_start >= 2500, 0,
+                                                (2500 - exp_tmrel.bw_start)/(exp_tmrel.bw_end - exp_tmrel.bw_start)))
+    tmrel_preterm_prevalence = (exp_tmrel.exposure * exp_tmrel.frac_preterm).sum()
+    tmrel_lbw_prevalence = (exp_tmrel.exposure * exp_tmrel.frac_lbw).sum()
+    rr_target = rr_data.loc[(rr_data.exposure == exposure_level)][draw].values[0]
```

Comment:
```suggestion
    rr_target = rr_data.loc[(rr_data.exposure == exposure_level)][draw].iloc[0]
```

(I recently learned from @NathanielBlairStahn that `values` is deprecated.)